server/api.go

```diff
@@ -61,216 +112,178 @@ func (p *Plugin) ServeHTTP(c *plugin.Context, w http.ResponseWriter, r *http.Req
-		"##### Notifications\n"+
-		"When someone mentions you, requests your review, comments on or modifies one of your pull requests/issues, or assigns you, you'll get a post here about it.\n"+
-		"Turn off notifications with `/github settings notifications off`.\n\n"+
-		"##### Sidebar Buttons\n"+
-		"Check out the buttons in the left-hand sidebar of Mattermost.\n"+
-		"* The first button tells you how many pull requests you have submitted.\n"+
-		"* The second shows the number of PR that are awaiting your review.\n"+
-		"* The third shows the number of PR and issues your are assiged to.\n"+
-		"* The fourth tracks the number of unread messages you have.\n"+
-		"* The fifth will refresh the numbers.\n\n"+
-		"Click on them!\n\n"+
-		"##### Slash Commands\n"+
-		strings.Replace(COMMAND_HELP, "|", "`", -1), gitUser.GetLogin(), gitUser.GetHTMLURL())
-	p.CreateBotDMPost(state.UserID, message, "custom_git_welcome")
+		if err := p.storeGitHubToUserIDMapping(gitUser.GetLogin(), state.UserID); err != nil {
+			fmt.Println(err.Error())
+		}
 
-	config := p.getConfiguration()
+		// Post intro post
+		message := fmt.Sprintf("#### Welcome to the Mattermost GitHub Plugin!\n"+
+			"You've connected your Mattermost account to [%s](%s) on GitHub. Read about the features of this plugin below:\n\n"+
+			"##### Daily Reminders\n"+
+			"The first time you log in each day, you will get a post right here letting you know what messages you need to read and what pull requests are awaiting your review.\n"+
+			"Turn off reminders with `/github settings reminders off`.\n\n"+
+			"##### Notifications\n"+
+			"When someone mentions you, requests your review, comments on or modifies one of your pull requests/issues, or assigns you, you'll get a post here about it.\n"+
+			"Turn off notifications with `/github settings notifications off`.\n\n"+
+			"##### Sidebar Buttons\n"+
+			"Check out the buttons in the left-hand sidebar of Mattermost.\n"+
+			"* The first button tells you how many pull requests you have submitted.\n"+
+			"* The second shows the number of PR that are awaiting your review.\n"+
+			"* The third shows the number of PR and issues your are assiged to.\n"+
+			"* The fourth tracks the number of unread messages you have.\n"+
+			"* The fifth will refresh the numbers.\n\n"+
+			"Click on them!\n\n"+
+			"##### Slash Commands\n"+
+			strings.Replace(COMMAND_HELP, "|", "`", -1), gitUser.GetLogin(), gitUser.GetHTMLURL())
+		p.CreateBotDMPost(state.UserID, message, "custom_git_welcome")
+
+		config := p.getConfiguration()
+
+		p.API.PublishWebSocketEvent(
+			WS_EVENT_CONNECT,
+			map[string]interface{}{
+				"connected":           true,
+				"github_username":     userInfo.GitHubUsername,
+				"github_client_id":    config.GitHubOAuthClientID,
+				"enterprise_base_url": config.EnterpriseBaseURL,
+				"organization":        config.GitHubOrg,
+			},
+			&model.WebsocketBroadcast{UserId: state.UserID},
+		)
+
+		html := `
+	<!DOCTYPE html>
```

Comment:
nit: Here you have one indentation too many for this literal block.

Reply:
Do you mean this?
```go
html := `
	<!DOCTYPE html>
	<html>
		<head>
			<script>
				window.close();
			</script>
		</head>
		<body>
			<p>Completed connecting to GitHub. Please close this window.</p>
		</body>
	</html>
			`
```
or I should remove indications inside string literal

Reply:
The ones inside the string literal.